fiano: init at 1.2.0

jmbaur opened the pull request:
###### Description of changes

Fiano is a suite of Go-based tools for modifying UEFI firmware.

This initial inclusion of fiano omits one package called `fittool`, since this
tool has a separate `modRoot` than the rest and its `go.sum` is not up to date,
so it does not build. I will be working with upstream to get this fixed.

I also added myself as a maintainer of this package, and thus adding myself to
the maintainer list as well.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See
      [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
    (look inside
    [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or
    [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in
    [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or
    [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are
    [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking)
    to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using
      `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all
      changes have to be committed, also see
      [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in
      `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or
        breaking
  - [ ] (Module updates) Added a release notes entry if the change is
        significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS
        module
- [x] Fits
      [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

natsukium commented on pkgs/tools/misc/fiano/default.nix:
```diff
@@ -0,0 +1,36 @@
+    description = "Go-based tools for modifying UEFI firmware";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ jmbaur ];
+    platforms = platforms.all;
```
```suggestion
```
`platforms.all` is the default for `buildGoModule`.

natsukium commented on pkgs/tools/misc/fiano/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  ldflags = [ "-s" "-w" ];
+
+  meta = with lib; {
```
It would be better to add meta.changelog.

jmbaur replied:
added

jmbaur committed:
fiano: init at 1.2.0

Fiano is a suite of Go-based tools for modifying UEFI firmware.

This initial inclusion of fiano omits one package called `fittool`,
since this tool has a separate `modRoot` than the rest and its `go.sum`
is not up to date, so it does not build. I will be working with upstream
to get this fixed.

jmbaur committed:
fiano: add jmbaur as maintainer

jmbaur commented on maintainers/maintainer-list.nix:
it is now in two commits

jmbaur commented on pkgs/tools/misc/fiano/default.nix:
```diff
@@ -0,0 +1,36 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
done

natsukium commented on pkgs/tools/misc/fiano/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
```
```suggestion
}:

buildGoModule rec {
```
Just a trivial point, but could you please add a blank line here?

natsukium reviewed (commented):
It looks much better now.
It would be better to remove the "v" from "v1.2.0" in the commit message.

jmbaur commented:
@natsukium thanks for the review! Let me know if there's anything else that looks off

natsukium commented:
Result of `nixpkgs-review pr 232715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiano</li>
  </ul>
</details>

natsukium reviewed (approved):
LTGM

pbek commented:
1 package built:
fiano

Result of `nixpkgs-review pr 232715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiano</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2275

SuperSandro2000 commented on pkgs/tools/misc/fiano/default.nix:
```diff
@@ -0,0 +1,40 @@
+}:
+
+buildGoModule rec {
+  pname = "fiano";
+
+  version = "1.2.0";
```
```suggestion
  pname = "fiano";
  version = "1.2.0";
```
